src/daft-local-execution/src/sinks/sort.rs

```diff
@@ -3,30 +3,22 @@ use std::sync::Arc;
 use common_error::DaftResult;
 use daft_dsl::ExprRef;
 use daft_micropartition::MicroPartition;
-use futures::{stream, StreamExt};
 use tracing::instrument;
 
-use crate::sources::source::Source;
-
 use super::blocking_sink::{BlockingSink, BlockingSinkStatus};
 
 pub struct SortSink {
     sort_by: Vec<ExprRef>,
     descending: Vec<bool>,
-    state: SortState,
-}
-
-enum SortState {
-    Building(Vec<Arc<MicroPartition>>),
```

Comment:
I would keep the state as an enum here since it's meant to be the abstract when we have partial results operators